### ISSUE TITLE
too many errors? (go 1.20.3)

### ISSUE THREAD

**Issue by marcelloh** [sugawarayuuta/sonnet#3]
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:10:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:12:6: missing function body
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:17:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:19:6: missing function body
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:24:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:26:6: missing function body
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:31:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:37:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:44:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:51:3: //go:linkname must refer to declared function or variable
vendor/github.com/sugawarayuuta/sonnet/internal/types/link.go:26:6: too many errors

**Comment by sugawarayuuta** [comment]
Hi! Thank you for the issue. This is caused by a change (fix?) in go 1.20 it seems. Internal issue is that I'm using linknames to methods (receiver functions) like so https://github.com/sugawarayuuta/sonnet/blob/main/internal/types/link.go#L8-L52 , and it's probably easily fixable

**Comment by marcelloh** [comment]
it is usable :-)

On my machine a small benchmark:
```
Benchmark_Decoding_Sonic
Benchmark_Decoding_Sonic-8    	 1307377	       910.1 ns/op	    1088 B/op	       9 allocs/op
Benchmark_Decoding_Sonnet
Benchmark_Decoding_Sonnet-8    	 1292158	       904.9 ns/op	    1088 B/op	       9 allocs/op
Benchmark_Encoding_Sonic
Benchmark_Encoding_Sonic-8    	 5390101	       208.6 ns/op	     112 B/op	       2 allocs/op
Benchmark_Encoding_Sonnet
Benchmark_Encoding_Sonnet-8    	 9270050	       129.1 ns/op	      96 B/op	       3 allocs/op
```
